R/db.R

```diff
@@ -16,7 +16,33 @@ table_name <- function(prefix, salt) {
 #' @param dbconn SQLConnection. A database connection.
 #' @importFrom DBI dbGetQuery
 column_names_map <- function(dbconn) {
-  DBI::dbGetQuery(dbconn, 'SELECT * FROM column_names')
+  DBI::dbGetQuery(dbconn, "SELECT * FROM column_names")
+}
+
+#' Fetch all the shards for the given table name.
+#'
+#' @name get_shards_for_table
+#' @param dbconn SQLConnection. A database connection.
+#' @param tbl_name character. The calculated table name for the function.
+#' @return one or many names of the shard tables.
+#' @importFrom DBI dbGetQuery
+get_shards_for_table <- function(dbconn, tbl_name) {
+  if (!DBI::dbExistsTable(dbconn, 'table_shard_map')) create_shards_table(dbconn, 'table_shard_map')
+  DBI::dbGetQuery(dbconn, paste0("SELECT shard_name FROM table_shard_map where table_name='", tbl_name, "'"))
+}
+
+#' Fetch all the shards for the given table name.
+#'
+#' @name create_table
```

Comment:
I think you mean `@rdname`. It should almost never be necessary to use `@name`, since that is determined by the function name below.